mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt

```diff
@@ -8,129 +8,42 @@ import android.view.View
-    @Inject
-    var mPresenter: BankAccountsPresenter? = null
-    var mBankAccountsPresenter: BankContract.BankAccountsPresenter? = null
-
-    @JvmField
-    @BindView(R.id.rv_accounts)
-    var mRvLinkedBankAccounts: RecyclerView? = null
-
-    @JvmField
-    @BindView(R.id.iv_empty_no_transaction_history)
-    var ivTransactionsStateIcon: ImageView? = null
-
-    @JvmField
-    @BindView(R.id.tv_empty_no_transaction_history_title)
-    var tvTransactionsStateTitle: TextView? = null
-
-    @JvmField
-    @Inject
-    var mBankAccountsAdapter: BankAccountsAdapter? = null
-
-    @JvmField
-    @BindView(R.id.linked_bank_account_text)
-    var linkedAccountsText: TextView? = null
-
-    @JvmField
-    @BindView(R.id.tv_empty_no_transaction_history_subtitle)
-    var tvTransactionsStateSubtitle: TextView? = null
+    lateinit var binding: FragmentAccountsBinding
+    //hilt viewmodel
+    private val viewModel: BankAccountsViewModel by viewModels()
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        val rootView = inflater.inflate(R.layout.fragment_accounts, container, false)
-        ButterKnife.bind(this, rootView)
-        setupRecycletView()
-        setUpSwipeRefresh()
-        mPresenter!!.attachView(this)
-        showSwipeProgress()
-        mBankAccountsPresenter!!.fetchLinkedBankAccounts()
-        return rootView
-    }
-
-    private fun setUpSwipeRefresh() {
-        swipeRefreshLayout?.setOnRefreshListener { mBankAccountsPresenter!!.fetchLinkedBankAccounts() }
-    }
-
-    private fun setupRecycletView() {
-        val layoutManager = LinearLayoutManager(context)
-        layoutManager.orientation = LinearLayoutManager.VERTICAL
-        mRvLinkedBankAccounts!!.layoutManager = layoutManager
-        mRvLinkedBankAccounts!!.setHasFixedSize(true)
-        mRvLinkedBankAccounts!!.adapter = mBankAccountsAdapter
-        mRvLinkedBankAccounts!!.addItemDecoration(
-            DividerItemDecoration(
-                context,
-                DividerItemDecoration.VERTICAL
-            )
-        )
-        mRvLinkedBankAccounts!!.addOnItemTouchListener(
-            RecyclerItemClickListener(
-                activity,
-                object : RecyclerItemClickListener.OnItemClickListener {
-                    override fun onItemClick(childView: View?, position: Int) {
-                        val intent = Intent(
-                            activity,
-                            BankAccountDetailActivity::class.java
-                        )
-                        intent.putExtra(
-                            Constants.BANK_ACCOUNT_DETAILS,
-                            mBankAccountsAdapter!!.getBankDetails(position)
-                        )
-                        intent.putExtra(Constants.INDEX, position)
-                        startActivityForResult(intent, BANK_ACCOUNT_DETAILS_REQUEST_CODE)
-                    }
```

Comment:
and have you handled this case in the new compose UI ?

Reply:
oops i missed this , would implement it on next commit along with your comment and desighn and binding .
Thanks ....